packages/webview_flutter/webview_flutter_wkwebview/darwin/Tests/HTTPCookieStoreProxyAPITests.swift

```diff
@@ -0,0 +1,68 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import WebKit
+import XCTest
+
+@testable import webview_flutter_wkwebview
+
+class HTTPCookieStoreProxyAPITests: XCTestCase {
+  @MainActor func testSetCookie() {
+    let registrar = TestProxyApiRegistrar()
+    let api = registrar.apiDelegate.pigeonApiWKHTTPCookieStore(registrar)
+
+    var instance: TestCookieStore? = TestCookieStore.customInit()
+    let cookie = HTTPCookie(properties: [
+      .name: "foo", .value: "bar", .domain: "http://google.com", .path: "/anything",
+    ])!
+
+    let expect = expectation(description: "Wait for setCookie.")
+    api.pigeonDelegate.setCookie(pigeonApi: api, pigeonInstance: instance!, cookie: cookie) {
+      result in
+      switch result {
+      case .success(_):
+        expect.fulfill()
+      case .failure(_):
+        break
+      }
+    }
+
+    wait(for: [expect], timeout: 1.0)
+    XCTAssertEqual(instance!.setCookieArg, cookie)
+
+    // Ensure instance is deallocated on main thread.
+    DispatchQueue.main.async {
+      instance = nil
+    }
+  }
+}
+
+class TestCookieStore: WKHTTPCookieStore {
+  var setCookieArg: HTTPCookie? = nil
+
+  // Workaround to subclass an Objective-C class that has an `init` constructor with NS_UNAVAILABLE
+  static func customInit() -> TestCookieStore {
+    let instance =
+      TestCookieStore.perform(NSSelectorFromString("new")).takeRetainedValue() as! TestCookieStore
+    return instance
+  }
+
+  #if compiler(>=6.0)
```

Comment:
`@MainActor` was required for local development with Xcode 16, but our CI doesn't recognize the method because it uses XCode 15. Related to: https://github.com/flutter/flutter/issues/148870

Reply:
Huh, I thought Swift 6 mode, with strict concurrency checking, was opt-in for Xcode 16.

Reply:
Yea, it looks like it is: https://developer.apple.com/documentation/swift/adoptingswift6#Decide-when-to-upgrade-to-strict-checking

I don't think I changed anything when I updated to Xcode 16.1. I originally had Swift 6, but changed it to 5 once I realized our CI wasn't using it. 

@cbracken / @vashworth When https://github.com/flutter/flutter/issues/148870 lands, will our CI continue to use Swift 5? Is it better practice for this plugin to support both Swift 6 & Swift 5?